frontend/src/metabase/reference/segments/SegmentQuestions.jsx

```diff
@@ -118,4 +125,14 @@ class SegmentQuestions extends Component {
   }
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(SegmentQuestions);
+export default _.compose(
+  Questions.loadList({
```

Comment:
:x:

`entityListLoader` is deprecated, so I'd say `entity.loadList` is [deprecated as well](https://github.com/metabase/metabase/blob/a989649ffd3310b14b7fcd694f25b9e3fb02c14a/frontend/src/metabase/entities/containers/index.js#L27-L28).

We should use `useQuestionListQuery` here and update `CardFilterOption` type.

Bonus points for marking `entity.loadList` as deprecated.

Reply:
Two points:
- `useQuestionListQuery` most likely has a bug and doesn't work with filter `f`
- even if it did work, you can't use a hook with a class component, and converting this to the functional component is useless since this whole UI will soon be removed

I spoke to @ranquild yesterday and he confirmed to not bother with the conversion to the functional component.

Reply:
If converting to a functional component is easy - let’s do it, otherwise that’s not that important here imho

Reply:
Done!
And the `useQuestionListQuery` works! 🚀 

Thanks for the encouragement to go down this route.